model/project_parser.go

```diff
@@ -585,43 +590,58 @@ func LoadProjectInto(ctx context.Context, data []byte, opts GetProjectOpts, iden
 	if p != nil {
 		*project = *p
 	}
+	catcher.Wrapf(err, LoadProjectError)
 	project.Identifier = identifier
-	return intermediateProject, errors.Wrap(err, LoadProjectError)
+	return intermediateProject, catcher.Resolve()
 }
 
+const (
```

Comment:
This is a good strategy! I think it's worth being more descriptive though, so it's clear looking back what's happening here. "Type" should maybe be called something more descriptive (RequestSource? ReadFileFrom? Not actually sure) or add a comment to Type to explain (also add a comment before the constants themselves).

Reply:
i changed it to ReadFileFrom

Reply:
Can we update these constants to make more sense for that? ReadFileFrom "mainline" doesn't make as much sense as "github" (local still also probably makes sense, i'm not actually sure what Patch should look like yet but we can push that off until the relevant ticket)

Reply:
done